packages/react/src/use-chat.ts

```diff
@@ -522,18 +525,62 @@ By default, it's set to 0, which will disable the feature.
 
       event?.preventDefault?.();
 
+      let files: MessageFile[] = [];
+
+      if (options.experimental_files) {
+        if (options.experimental_files instanceof FileList) {
+          for (const file of Array.from(options.experimental_files)) {
+            const { name, type } = file;
+
+            const dataUrl = await new Promise<string>((resolve, reject) => {
+              const reader = new FileReader();
+              reader.onload = readerEvent => {
+                resolve(readerEvent.target?.result as string);
+              };
+              reader.onerror = error => reject(error);
+              reader.readAsDataURL(file);
+            });
+
+            files.push({
+              name,
+              mimeType: type,
+              url: dataUrl,
+            });
+          }
+        } else if (Array.isArray(options.experimental_files)) {
+          for (const file of options.experimental_files) {
+            const { name, url, mimeType } = file;
+
+            files.push({
+              name,
+              mimeType,
+              url,
+            });
+          }
+        } else {
+          throw new Error('Invalid files type');
+        }
+      }
+
       const requestOptions = {
         headers: options.headers ?? options.options?.headers,
         body: options.body ?? options.options?.body,
       };
 
       const chatRequest: ChatRequest = {
         messages: input
-          ? messagesRef.current.concat({
-              id: generateId(),
-              role: 'user',
-              content: input,
-            })
+          ? options.experimental_files
+            ? messagesRef.current.concat({
+                id: generateId(),
+                role: 'user',
+                content: input,
+                experimental_files: files,
+              })
+            : messagesRef.current.concat({
+                id: generateId(),
+                role: 'user',
+                content: input,
+              })
```

Comment:
would it be sufficient to just add `experimental_files: options.experimental_files`?

Reply:
not sure if I understand correctly but `options.experimental_files` and `files` are not equivalent, I've renamed them to `attachmentsFromOptions` and `attachmentsForRequest` for better clarity

the attachments are converted to data urls when they're of type `FileList`